console_backend/src/ntrip_output.rs

```diff
@@ -0,0 +1,109 @@
+use crate::ntrip_tab::OutputType;
+use crate::types::{ArcBool, Result};
+use anyhow::Context;
+use crossbeam::channel::Receiver;
+use log::error;
+use std::io::Write;
+use std::process::{Command, Stdio};
+use std::{io, thread};
+
+pub struct MessageConverter {
+    in_rx: Receiver<Vec<u8>>,
+    running: ArcBool,
+    output_type: OutputType,
+}
+
+impl MessageConverter {
+    pub fn new(in_rx: Receiver<Vec<u8>>, output_type: OutputType) -> Self {
+        Self {
+            in_rx,
+            running: ArcBool::new(),
+            output_type,
+        }
+    }
+
+    pub fn start<W: Write + Send + 'static>(&mut self, out: W) -> Result<()> {
+        self.running.set(true);
+        match self.output_type {
+            OutputType::RTCM => self.output_rtcm(out),
+            OutputType::SBP => self.output_sbp(out),
+        }
+    }
+
+    /// Just redirects directly to writer
+    fn output_rtcm<W: Write + Send + 'static>(&mut self, mut out: W) -> Result<()> {
+        let in_rx = self.in_rx.clone();
+        let running = self.running.clone();
+        thread::spawn(move || loop {
+            if !running.get() {
+                break;
+            }
+            if let Ok(data) = in_rx.try_recv() {
+                if let Err(e) = out.write(&data) {
+                    error!("failed to write to device {e}");
+                }
+            }
+        });
+        Ok(())
+    }
+
+    /// Runs rtcm3tosbp converter
+    fn output_sbp<W: Write + Send + 'static>(&mut self, mut out: W) -> Result<()> {
+        let curr_dir = std::env::current_dir()?;
+        let curr_dir = curr_dir.to_str().expect("current directory not found");
+        let cmd = format!("ls {curr_dir}");
+        println!("{}", cmd);
```

Comment:
```
ls /

Applications
Library
System
Users
Volumes
bin
cores
dev
etc
home
opt
private
sbin
tmp
usr
var
```
not relative to `.app`, 

this pr uses selector instead of bundling
https://github.com/swift-nav/swift-toolbox/pull/1076

Reply:
hmm am stuck on whether there is an easy way to locate the path for bundled files i.e. `Swift Console.app\Contents\...` on mac

alternatively really simple, [here is a test released version](https://github.com/swift-nav/swift-toolbox/pull/1076) with user provided binary (to save the packaging trouble)

or maybe like - `include_bytes`, when ran write the bytes to file to execute.

or maybe pass path from python? https://github.com/swift-nav/swift-toolbox/blob/3f4874edafdae9a55073065bdf13885e536c953e/swiftnav_console/main.py#L584

any ideas? 🤔 
@silverjam @swift-nav/devinfra

Reply:
https://doc.rust-lang.org/std/env/fn.current_exe.html

This returns a PathBuf, wouldn't that be sufficient to locate a helper binary?

Reply:
> https://doc.rust-lang.org/std/env/fn.current_exe.html
> 
> This returns a PathBuf, wouldn't that be sufficient to locate a helper binary?

yeah that returns
```
"/Users/adriankong/IdeaProjects/v/swift-toolbox/py39/bin/python3"
```
path to python on build oddly enough

Reply:
We've already solved this problem for other things (like windowpos exe and qt web engine process) we should be able to adapt one of those existing solutions.

Reply:
> not relative to `Swift Console.app` when ran in release on mac

The current working directory of the app is going to vary widely depending on how it's launched, so this isn't a viable solution.

Reply:
> It's something that's going to need to be handled differently if we're running a development build versus a packaged build.
> 
> Similar problems are handled by our `entrypoint` code:
> 
> https://github.com/swift-nav/swift-toolbox/blob/3f4874edafdae9a55073065bdf13885e536c953e/entrypoint/src/main.rs#L66

I'll give this one a try 👍